internal/docker/client.go

```diff
@@ -116,29 +118,30 @@ type Cli struct {
 	env       Env
 }
 
-func NewDockerClient(ctx context.Context, env Env) (*Cli, error) {
+func NewDockerClient(ctx context.Context, env Env) Client {
 	opts, err := CreateClientOpts(ctx, env)
 	if err != nil {
-		return nil, errors.Wrap(err, "CreateClientOpts")
+		return newExplodingClient(err)
```

Comment:
hm, is it it not worth `errors.Wrap`s throughout to add context to the client explosion?